### PR TITLE
Use static ids in Poedit 3.5+

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Related to https://github.com/vslavik/poedit/pull/850

### Summary of the issue:
The Poedit appmodule relies on some complex calculation logic to announce several pieces of information with shortcuts, e.g. translator notes/comments/warnings.

### Description of user facing changes
None

### Description of development approach
@vslavik kindly introduced static ids for the controls we need for our appModule. That work is in https://github.com/vslavik/poedit/pull/850

### Testing strategy:
Test with https://download.poedit.net/tmp/Poedit-3.5nvda-setup.exe

- [x] Translation warnings
- [x] Commments
- [x] Old source text
- [x] Translator notes
- [x] Pro version
- [x] Different languages with one or more plural forms

### Known issues with pull request:
Breaks compatibility with Poedit versions before this fix, but should stay compatible with all versions to come, as long as these ids stay static.

### Code Review Checklist:

- [ ] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [ ] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
-x[ ] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Consolidated control ID management into a single class for improved maintainability and clarity.
	- Streamlined method signatures for retrieving objects associated with control IDs.

- **Bug Fixes**
	- Refined string formatting in the `CancellableSpeech` class to improve output consistency.

- **Refactor**
	- Removed redundant classes and updated method implementations to enhance code readability and structure.

### CODE REVIEW

**Comment by LeonarddeR** [comment]
@ruifontes I recall you have the pro version. Would you be able to test with this build of poedit along with this pr?

**Comment by CyrilleB79** [comment]
@LeonarddeR in addition, would you test strings with plurals for:
* languages with 2 forms (e.g. English); probably already done
* language with only one form (e.g. Chinese)
* language with more than 2 forms, e.g. Slovenian.

**Comment by vslavik** [comment]
> @LeonarddeR in addition, would you test strings with plurals for:

Ah, this is something I meant to ask: do you need static IDs for some or all of the text fields (source plural + singular; translation singular or translation plurals (variable count))?

**Comment by LeonarddeR** [comment]
I don't think we need these now, but people might have different opinions about that

**Comment by vslavik** [comment]
> I don't think we need these now, but people might have different opinions about that

OK, so let's only add fixed IDs when they become useful for something (feel free to e.g. add comment there to file an issue at poedit in such cases).

**Comment by CyrilleB79** [comment]
If the first compatible poedit version changes with this PR, it should be mentioned in the change log.

**Comment by LeonarddeR** [comment]
The first compatible version is not yet released. @vslavik is not sure about its release date.
That said, the milestone for 2024.4 is due october 28, I guess it is safe to assume that version 3.5 will be release before then.

**Comment by CyrilleB79** [comment]
This PR can be reviewed but it should not be merged until poedit 3.5 is available.

**Comment by LeonarddeR** [comment]
@seanbudd This pr needs to go in the earliest possible release after the release of Poedit 3.5, which release date is unfortunately not yet known.

**Comment by vslavik** [comment]
> Poedit 3.5, which release date is unfortunately not yet known.

August 30.

**Comment by SaschaCowley** [comment]
@LeonarddeR will Poedit 3.5 support be broken without this PR?

File: source/appModules/poedit.py [hunk 1]
```diff
@@ -29,31 +27,31 @@
...
+	"""The "Needs work" toggle in editing area at the bottom"""
+	TRANSLATION_ISSUE_TEXT = 10102
+	"""
+	The error or warning line above translation field
+	(hidden when there's no issue; ID is of the static text child window with issue's text)
+	"""
+	PREVIOUS_SOURCE_TEXT = 10103
+	"""
+	Text of previous source text
+	(msgid) for current item(shown in sidebar, may be hidden, is static control with the text)
```
Comment (seanbudd) [inline source/appModules/poedit.py:43]:
```suggestion
	(msgid) for current item (shown in sidebar, may be hidden, is static control with the text)
```

File: source/appModules/poedit.py [hunk 2]
```diff
@@ -29,31 +27,31 @@
...
-	Therefore, using a wxDataView control in the translations list as a reference,
-	we can safely calculate control ids accross releases or instances.
-	This class contains window control id offsets relative to the wxDataView window.
-	"""
-
-	PRO_IDENTIFIER = -10  # This is a button in the free version
-	MAIN_SPLITTER_IDENTIFIER = -2  # The splitter that holds the translation list
-	TRANSLATION_WARNING = 17
-	NEEDS_WORK_SWITCH = 21
+class _WindowControlId(IntEnum):
```
Comment (seanbudd) [inline source/appModules/poedit.py:30]: could members of this enum get new lines to separate them for readability?

**Review by seanbudd — COMMENTED** [review]
Generally looks good to me

**Comment by LeonarddeR** [comment]
> @LeonarddeR will Poedit 3.5 support be broken without this PR?

Yes.

File: user_docs/en/changes.md [hunk 3]
```diff
@@ -24,6 +24,7 @@ The available options are:
 * eSpeak NG has been updated to 1.52-dev commit `961454ff`. (#16775)
   * Added new languages Faroese and Xextan.
 * When using a multi-line braille display via the standard HID braille driver, all lines of cells will be used. (#16993, @alexmoon)
+* The stability of NVDA's Poedit support has been improved with the side effect that the minimum required version of Poedit is now version 3.5. (#16889, @LeonarddeR)
```
Comment (seanbudd) [inline user_docs/en/changes.md:27]: can something be added to the release blurb to note this?

File: user_docs/en/changes.md [hunk 4]
```diff
@@ -18,6 +18,8 @@ Cursor routing reliability has been improved, and support for routing keys in Po
 All lines of cells will now be used when using a multi-line braille display via HID braille.
 NVDA is no longer unstable after restarting NVDA during an automatic Braille Bluetooth scan.
 
+Due to stability updates in the poedit translation software and changes in NVDA to support these, the minimum required version of Poedit that works with NVDA is now version 3.5.
```
Comment (seanbudd) [inline user_docs/en/changes.md:21]:
```suggestion
Due to stability updates in the Poedit translation software and changes in NVDA to support these, the minimum required version of Poedit that works with NVDA is now version 3.5.
```

**Review by seanbudd — COMMENTED** [review]
note changes to liblouis have been pushed

**Comment by LeonarddeR** [comment]
@seanbudd wrote:
> note changes to liblouis have been pushed

How are those changes related to this pr?

**Comment by seanbudd** [comment]
@LeonarddeR - you may have committed something by mistake  it was pushed in e91216d1cf607a7f2ae222c1b3487167fcf2e2f5

**Comment by LeonarddeR** [comment]
I'm sorry. Should've been fixed now.

**Review by seanbudd — APPROVED** [review]

**Comment by LeonarddeR** [comment]
[Poedit 3.5 is there](https://poedit.net/news/poedit-3.5-excel-and-blade/)!

**Review by michaelDCurran — APPROVED** [review]

**Review by SaschaCowley — CHANGES_REQUESTED** [review]
Please edit the Poedit section of the User Guide to state version 3.5 is the minimum supported version.

**Review by SaschaCowley — APPROVED** [review]

**Pull request merged**